m4/curl-rustls.m4

```diff
@@ -88,6 +88,11 @@ if test "x$OPT_RUSTLS" != xno; then
             CPPFLAGS="$CPPFLAGS $addcflags"
         fi
 
+        if test $(uname) = "Darwin"; then
+            RUSTLS_LDFLAGS="-framework Security -framework Foundation"
+        else
+            RUSTLS_LDFLAGS="-lpthread -ldl -lm"
+        fi
```

Comment:
```suggestion
        case $host in
          *-apple-*)
            RUSTLS_LDFLAGS="-framework Security -framework Foundation"
          *)
            RUSTLS_LDFLAGS="-lpthread -ldl -lm"
        esac
```
Would it make sense to assume this works for all Apple platforms, not just macOS, and update the check accordingly? Also the logic at line 39 (adding `Security` to `LDFLAGS`, even if rustls isn't found?) is likely unnecessary after this.

Reply:
Thanks! I adjusted the script as you suggested and removed the initial LDFLAGS setting.